### PR TITLE
feat(gta/core): dynamic pool resizing

### PR DESCRIPTION
**Pull request by duk-37**
A very preliminary attempt at abusing virtual memory to allow for dynamically-"resizable" pools. At the moment eligible pools need to be manually specified with a hash and size/growth policies.

Can be enabled by passing `-poolhacks` as a command line option.

Note that there's still a fixed upper limit, but it can be raised much higher because the actual memory is only committed on an as-needed basis.

Ideally we get this working with the `TxdStore` pool but that needs me to either:
1. Hardcode its location somehow
2. Figure out how to exclude pre-allocated pools from pool hackery and have reasonable defaults

### CODE REVIEW

**Comment by donkras** [comment]
duk doing stuff again!!!! no way

**Comment by duk-37** [comment]
🙂 
image

**Comment by duk-37** [comment]
note: above is not fully working yet, streaming stuff also needs hashmap resize stuff

**Comment by antigonos7** [comment]
> note: above is not fully working yet, streaming stuff also needs hashmap resize stuff

2802 has little space for extra txd
Is it possible to have 105500 txd poolsize?

**Comment by duk-37** [comment]
sincere congratulations to everyone on the fivem team :)

File: code/components/gta-core-five/src/PoolManagement.cpp [hunk 1]
```diff
@@ -237,15 +401,131 @@ static void* PoolAllocateWrap(atPoolBase* pool)
...
+
+					// populate new free entries in hashmap
+					if (nextFree == -1)
+					{
+						assetMap->nextFreeEntry = currentSize;
+						nextFree = currentSize;
+					}
+					else
+					{
+						assetMap->data[nextFree].link = currentSize;
```
Comment (duk-37) [inline code/components/gta-core-five/src/PoolManagement.cpp:508]: todo: pretty sure this is wrong,`nextFree = currentSize;` needs to be on both branches

**Comment by Legacy-TacticalGamingInteractive** [comment]
:(

**Comment by prikolium-cfx** [comment]
Thanks, duk. We appreciate your contribution and passion, but we decided to implement it in another way.

**Pull request closed without merging**